lib/spack/spack/cmd/__init__.py

```diff
@@ -177,6 +178,28 @@ def parse_specs(
     if not concretize:
         return specs
 
+    # Special case for concretizing a single spec
+    if len(specs) == 1:
+        return [specs[0].concretized()]
+
+    # Special case if every spec has an abstract spec
+    if all(spec.concrete or spec.abstract_hash for spec in specs):
+        ret = [s.concretized() for s in specs]
+
+        unify = spack.config.get("concretizer:unify", False)
+        if unify is True:  # True, "when_possible", False are possible values
+            specs_per_name = Counter(
+                spec.name
+                for spec in traverse.traverse_nodes(ret, deptype=("link", "run"))
+                if spec.name != "gcc-runtime"  # gcc runtime is special and allowed multiples
+            )
+            if any(count > 1 for count in specs_per_name.values()):
+                msg = "Specs conflict and `concretizer:unify` is configured true.\n"
+                msg += f"    specs: {specs}\n"
+                msg += "    Either loosen the spec constraints or change unification config."
```

Comment:
this could be better and could point out what duplicates are there -- I used this to debug the above problem.